### PR TITLE
HEXA-192 #Programmatically creating a new user and role for the client.

### PR DESCRIPTION
**Pull request by zJANOTTI**

### CODE REVIEW

File: web/profiles/custom/hexa/src/Form/ClientSettingsForm.php [hunk 1]
```diff
@@ -33,6 +33,14 @@ public function buildForm(array $form, FormStateInterface $form_state) {
     $form_state->disableCache();
     $form['#title'] = $this->t('Configure client preferences');
 
+    $form['name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Name'),
+      '#description' => $this->t('Name of your company to display for users.'),
+      '#default_value' => t('Hexentials'),
```
Comment (rafaolf) [inline web/profiles/custom/hexa/src/Form/ClientSettingsForm.php:40]: I believe we are missing the _$this_ instruction here as we are inside the class.

File: web/profiles/custom/hexa/src/Form/ClientSettingsForm.php [hunk 2]
```diff
@@ -150,6 +158,20 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
+    $name = (string) $form_state->getValue('name');
+    $mail = (String) $form_state->getValue('mail');
```
Comment (rafaolf) [inline web/profiles/custom/hexa/src/Form/ClientSettingsForm.php:162]: Will _String_ in uppercase work? Please adopt the casting pattern here.
**DOCS:** https://www.php.net/manual/en/language.types.type-juggling.php

File: web/profiles/custom/hexa/src/Form/ClientSettingsForm.php [hunk 3]
```diff
@@ -150,6 +158,20 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
...
+
+    $user = \Drupal\user\Entity\User::create();
+    $user->setPassword((strtolower($name)));
+    $user->enforceIsNew();
+    $user->setEmail($mail);
+    $user->setUsername((strtolower($name)));
+    $user->set("langcode", $language);
+    $user->addRole('content_manager');
+    $user->activate();
+    $user->save();
```
Comment (rafaolf) [inline web/profiles/custom/hexa/src/Form/ClientSettingsForm.php:173]: Can we please create a function here to keep things organized?
e.g.: _updateConfiguration()_;

File: web/profiles/custom/hexa/src/Form/ClientSettingsForm.php [hunk 4]
```diff
@@ -150,6 +158,20 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
...
+    $name = (string) $form_state->getValue('name');
+    $mail = (String) $form_state->getValue('mail');
+    $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
+
+    $user = \Drupal\user\Entity\User::create();
+    $user->setPassword((strtolower($name)));
+    $user->enforceIsNew();
+    $user->setEmail($mail);
+    $user->setUsername((strtolower($name)));
+    $user->set("langcode", $language);
```
Comment (rafaolf) [inline web/profiles/custom/hexa/src/Form/ClientSettingsForm.php:170]: We don't usually use double-quotes on Drupal's standards (only for very specific scenarios).

File: web/profiles/custom/hexa/src/Form/ClientSettingsForm.php [hunk 5]
```diff
@@ -150,6 +158,20 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
+    $name = (string) $form_state->getValue('name');
+    $mail = (String) $form_state->getValue('mail');
+    $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
```
Comment (rafaolf) [inline web/profiles/custom/hexa/src/Form/ClientSettingsForm.php:163]: Did you try to select portuguese here to see if that works? I'm not 100% sure that by the installation moment we will have the service working to provide us the correct language. If that works, please ignore this remark.

**Review by rafaolf — COMMENTED** [review]
Excellent work! Just simple remarks usually related to coding standards.

File: web/profiles/custom/hexa/src/Form/ClientSettingsForm.php [hunk 6]
```diff
@@ -4,8 +4,8 @@
 
 use Drupal\Core\Form\FormBase;
 use Drupal\Core\Form\FormStateInterface;
-use Drupal\file\Entity\File;
 use Drupal\user\UserInterface;
+use PhpParser\Node\Expr\Cast\String_;
```
Comment (rafaolf) [inline web/profiles/custom/hexa/src/Form/ClientSettingsForm.php:8]: I believe we don't need that.

File: web/profiles/custom/hexa/src/Form/ClientSettingsForm.php (hunk 2 above)
Comment (zJANOTTI) [reply web/profiles/custom/hexa/src/Form/ClientSettingsForm.php:162]: I don't know i wrote it in uppercase, anyways (string) is the correct way

File: web/profiles/custom/hexa/src/Form/ClientSettingsForm.php (hunk 5 above)
Comment (zJANOTTI) [reply web/profiles/custom/hexa/src/Form/ClientSettingsForm.php:163]: it works

File: web/profiles/custom/hexa/src/Form/ClientSettingsForm.php (hunk 4 above)
Comment (zJANOTTI) [reply web/profiles/custom/hexa/src/Form/ClientSettingsForm.php:170]: Corrected

File: web/profiles/custom/hexa/src/Form/ClientSettingsForm.php (hunk 6 above)
Comment (rafaolf) [inline web/profiles/custom/hexa/src/Form/ClientSettingsForm.php:8]: Why we need that? Please let me know.

**Pull request closed without merging**